services/ui/ws/server_window.cc

```diff
@@ -177,7 +177,28 @@ void ServerWindow::SetBounds(
     const base::Optional<cc::LocalSurfaceId>& local_surface_id) {
   if (bounds_ == bounds && current_local_surface_id_ == local_surface_id)
     return;
+  SetBoundsInternal(bounds, local_surface_id);
+}
+
+void ServerWindow::SetBoundsFromHostWindowManager(
+    const gfx::Rect& bounds,
+    const base::Optional<cc::LocalSurfaceId>& local_surface_id) {
+  if (bounds_ == bounds && current_local_surface_id_ == local_surface_id)
+    return;
 
+  gfx::Rect old_bounds = bounds_;
+  SetBoundsInternal(bounds, local_surface_id);
+  if (bounds.origin() != old_bounds.origin()) {
+    // Setting another origin to |root_| must be avoided when bounds are sent
```

Comment:
// The root_'s origin must stay at 0,0, so that events are correctly translated after bounds change.
// TODO(msisov): Maybe there is a better way than actually restore the original origin?